pkg/logql/log/consolidated_variant_extractor.go

```diff
@@ -74,23 +74,18 @@ func (c *consolidatedMultiVariantStreamExtractor) Process(ts int64, line []byte,
 }
 
 func appendVariantLabel(lbls LabelsResult, variantIndex int) LabelsResult {
-	streamLbls := lbls.Stream()
-
-	newLbls := make(labels.Labels, 0, len(streamLbls)+1)
-	newLbls = append(newLbls, labels.Label{
-		Name:  constants.VariantLabel,
-		Value: strconv.Itoa(variantIndex),
-	})
-	newLbls = append(newLbls, streamLbls...)
+	newLblsBuilder := labels.NewBuilder(lbls.Stream())
```

Comment:
You Spotify is the scratch builder here to avoid extra allocations.

Reply:
Changed in 4313bda.